### PR TITLE
Fixed `tid` set logic in MatrixGen

### PR DESCRIPTION
**Pull request by seansica**
In enterprise-attack v11.2, there are two `attack-pattern` STIX objects which trigger the unhandled exceptions described in Issue #76. They are as follows:
```
[
    "attack-pattern--970a3432-3237-47ad-bcca-7d8cbb217736",
    "attack-pattern--824add00-99a1-4b15-9a2d-6c5683b7b497"
]
```

Of note, the exceptions are not specific to the TAXII server. Setting the `source` parameter in `_get_technique_listing` to `local` instead of `source=taxii` and loading STIX direct from the [latest source](https://raw.githubusercontent.com/mitre-attack/attack-stix-data/master/enterprise-attack/enterprise-attack.json) still yields the same exceptions.

The first attack-pattern contains the following two edge cases. These are elements in the object's `external_references` array where `"[aA]ttack"` is a substring in their respective `source_name` properties.
```
{
		"source_name": "Github PSAttack", <--- HERE
		"description": "Haight, J. (2016, April 21). PS>Attack. Retrieved June 1, 2016.",
		"url": "https://github.com/jaredhaight/PSAttack"
}, 
{
		"source_name": "inv_ps_attacks", <--- HERE
		"description": "Hastings, M. (2014, July 16). Investigating PowerShell Attacks. Retrieved December 1, 2021.",
		"url": "https://powershellmagazine.com/2014/07/16/investigating-powershell-attacks/"
}
```

The second attack-pattern contains these two edge cases:
```
{
		"source_name": "inv_ps_attacks", <--- HERE
		"description": "Hastings, M. (2014, July 16). Investigating PowerShell Attacks. Retrieved December 1, 2021.",
		"url": "https://powershellmagazine.com/2014/07/16/investigating-powershell-attacks/"
}, 
{
		"source_name": "Praetorian TLS Downgrade Attack 2014", <--- HERE
		"description": "Praetorian. (2014, August 19). Man-in-the-Middle TLS Protocol Downgrade Attack. Retrieved October 8, 2021.",
		"url": "https://www.praetorian.com/blog/man-in-the-middle-tls-ssl-protocol-downgrade-attack/"
}
```
Common amongst all 4 edge cases is their non-zero index position in `external_references`.

Under certain conditions, `MatrixGen._get_technique_listing` attempts to read every element's `external_id` property when the string "_attack_" is detected in the element's `source_name` property. The problem is - this logic should only apply to the _first_ element in the `external_references` array. Currently, it is applying to _all_ elements.

Moreover, I found a tangentially related issue: the read operation for `tid` is only triggering if the parent objects' `kill_chain_phases[0]['kill_chain_name']` element is equal to string `"mitre-mobile"` or `"mitre-mobile-attack"`. `"mobile-attack"`, and `"mitre-ics-attack"` are also valid values for this element, so the reference activation list needs to expand from 2 to 4.

To summarize, there are two issues with detecting ATT&CK ID entries in `external_references`. 

Currently, the logic is:

- If it exists, the ATT&CK ID is stored anywhere in `external_references` (_ISSUE 1_)
- The ATT&CK ID entry always has a `source_name` with a value from: ['mitre-attack', 'mitre-mobile-attack'] (_ISSUE 2_)
- The ATT&CK ID entry always has the three properties: `source_name`, `external_id`, and `url`

But the logic should be:

- If it exists, the ATT&CK ID is stored  in `external_references[0]`
- The ATT&CK ID entry always has a `source_name` with a value from: ['mitre-attack', 'mitre-mobile-attack', 'mobile-attack', 'mitre-ics-attack']
- The ATT&CK ID entry always has the three properties: `source_name`, `external_id`, and `url`

### CODE REVIEW

File: mitreattack/navlayers/exporters/matrix_gen.py [hunk 1]
```diff
@@ -195,9 +196,8 @@ def _get_technique_listing(self, tactic, domain='enterprise'):
         techs = self._search(domain, [Filter('type', '=', 'attack-pattern'),
                                       Filter('kill_chain_phases.phase_name', '=', tactic)])
         for entry in techs:
-            if entry['kill_chain_phases'][0]['kill_chain_name'] == 'mitre-attack' or \
-                            entry['kill_chain_phases'][0]['kill_chain_name'] == 'mitre-mobile-attack':
-                tid = [t['external_id'] for t in entry['external_references'] if 'attack' in t['source_name']]
+            if entry['kill_chain_phases'][0]['kill_chain_name'] in MITRE_ATTACK_ID_SOURCE_NAMES:
+                tid = [t['external_id'] for t in entry['external_references'][:1] if 'attack' in t['source_name']]
```
Comment (jondricek) [inline mitreattack/navlayers/exporters/matrix_gen.py:200]: sometimes `attack` is in the `source_name` field for non-ATT&CK objects, so that isn't the best way to get an ATT&CK ID. we probably need to find a better way to guarantee that this will always only get ATT&CK IDs.

**Comment by jondricek** [comment]
@seansica are you still working on this?

File: mitreattack/navlayers/exporters/matrix_gen.py (hunk 1 above)
Comment (seansica) [reply mitreattack/navlayers/exporters/matrix_gen.py:200]: We will investigate ways to further optimize the ATT&CK ID detection process. For now, this PR should be merged as it resolves Issue #76 and still improves the aforementioned process.

**Comment by seansica** [comment]
This Issue (#78) and PR has been re-assessed and is ready to be merged.
